plugins/modules/helm_registry_auth.py

```diff
@@ -0,0 +1,235 @@
+  returned: always
+command:
+  type: str
+  description: Full C(helm) command executed
+  returned: always
+  sample: '/usr/local/bin/helm registry login oci-registry.domain.example --username=admin --password-stdin --insecure'
+failed:
+  type: bool
+  description: Indicate if the C(helm) command failed
+  returned: always
+  sample: false
+"""
+
+from ansible_collections.kubernetes.core.plugins.module_utils.helm import (
+    AnsibleHelmModule,
+)
+
+
+def arg_spec():
+    return dict(
+        binary_path=dict(type="path", required=False),
+        host=dict(type="str", aliases=["registry_url"], required=True),
+        state=dict(default="present", choices=["present", "absent"], required=False),
+        insecure=dict(type="bool", default=False, required=False),
+        username=dict(type="str", aliases=["repo_username"], required=False),
+        password=dict(
+            type="str", aliases=["repo_password"], no_log=True, required=False
+        ),
+        key_file=dict(type="path", required=False),
+        cert_file=dict(type="path", required=False),
+        ca_file=dict(type="path", required=False),
+    )
+
+
+def login(
+    command,
+    host,
+    insecure,
+    username,
+    password,
+    key_file,
+    cert_file,
+    ca_file,
+):
+    login_command = command + " registry login " + host
+
+    if username is not None and password is not None:
+        login_command += " --username=" + username + " --password-stdin"
+
+    if insecure:
+        login_command += " --insecure"
+
+    if key_file is not None:
+        login_command += " --key-file=" + key_file
+
+    if cert_file is not None:
+        login_command += " --cert-file=" + cert_file
+
+    if ca_file is not None:
+        login_command += " --ca-file=" + ca_file
+
+    return login_command
+
+
+def logout(command, host):
+    return command + " registry logout " + host
+
+
+def main():
+    global module
+
+    module = AnsibleHelmModule(
+        argument_spec=arg_spec(),
+        required_together=[["username", "password"]],
+        supports_check_mode=True,
+    )
+
+    changed = False
+
+    host = module.params.get("host")
+    state = module.params.get("state")
+    insecure = module.params.get("insecure")
+    username = module.params.get("username")
+    password = module.params.get("password")
+    key_file = module.params.get("key_file")
+    cert_file = module.params.get("cert_file")
+    ca_file = module.params.get("ca_file")
+
+    helm_cmd = module.get_helm_binary()
+
+    if state == "absent":
+        helm_cmd = logout(helm_cmd, host)
+        changed = True
+    elif state == "present":
+        helm_cmd = login(
+            helm_cmd, host, insecure, username, password, key_file, cert_file, ca_file
+        )
+        changed = True
+
+    if module.check_mode:
+        module.exit_json(changed=changed, command=helm_cmd)
+
+    rc, out, err = module.run_helm_command(helm_cmd, data=password)
```

Comment:
```suggestion
    rc, out, err = module.run_helm_command(helm_cmd, data=password, fails_on_error=False)
```
This is needed to ensure the idempotency of the log-out operation